### PR TITLE
support webhook debug api

### PR DESCRIPTION
**Pull request by buptcozy**
Signed-off-by: yangzhang <bupt_cozy@126.com>

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/webhook/api_debug_server.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"k8s.io/klog/v2"
+)
+
+var debugApiProviderMap = map[string]DebugApiProvider{}
+
+type DebugApiProvider interface {
+	RegisterDebugApi(engine *gin.Engine)
```
Comment (eahydra) [inline pkg/webhook/api_debug_server.go:31]: Consider not passing `gin.Engine` as a parameter directly, instead passing `gin.RouterGroup`

File: pkg/webhook/api_debug_server.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"net"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"k8s.io/klog/v2"
+)
+
+var debugApiProviderMap = map[string]DebugApiProvider{}
+
+type DebugApiProvider interface {
```
Comment (eahydra) [inline pkg/webhook/api_debug_server.go:30]: `API` should be all uppercase or all lowercase, to maintain a consistent style

File: pkg/webhook/api_debug_server.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		MaxHeaderBytes: 1 << 20,
+	}
+
+	for name, provider := range debugApiProviderMap {
+		provider.RegisterDebugApi(engine)
+		klog.Infof("Success register debug api handler, name:%v, tcpAddr:%v", name, tcpAddr)
+	}
+
+	go func() {
+		err := server.Serve(listener)
```
[the remark this answers is not in the document]
Comment (eahydra) [reply pkg/webhook/api_debug_server.go:74]: +1.

File: pkg/webhook/elasticquota/quota_topology.go [hunk 4]
```diff
@@ -168,3 +168,35 @@ func (qt *quotaTopology) fillQuotaDefaultInformation(quota *v1alpha1.ElasticQuot
 	}
 	return nil
 }
+
+type QuotaTopologySummary struct {
+	QuotaInfoMap       map[string]*core.QuotaInfoSummary
```
Comment (eahydra) [inline pkg/webhook/elasticquota/quota_topology.go:173]: Consider adding json tags?

File: pkg/webhook/elasticquota/validating/validating_handler.go [hunk 5]
```diff
@@ -125,3 +126,12 @@ func (h *ElasticQuotaValidatingHandler) InjectCache(cache cache.Cache) error {
 	})
 	return nil
 }
+
+func (h *ElasticQuotaValidatingHandler) RegisterDebugApi(engine *gin.Engine) {
+	routes := engine.Group("/apis/v1")
```
Comment (eahydra) [inline pkg/webhook/elasticquota/validating/validating_handler.go:131]: The `/apis/v1` URL prefix should be directly bound externally, and the provider only needs to consider which endpoints to support. If it's the DEBUG API, we might consider changing the name.

File: pkg/webhook/api_debug_server.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (buptcozy) [reply pkg/webhook/api_debug_server.go:74]: fixed

File: pkg/webhook/api_debug_server.go (hunk 2 above)
Comment (buptcozy) [reply pkg/webhook/api_debug_server.go:30]: fixed

File: pkg/webhook/elasticquota/quota_topology.go (hunk 4 above)
Comment (buptcozy) [reply pkg/webhook/elasticquota/quota_topology.go:173]: fixed

File: pkg/webhook/elasticquota/validating/validating_handler.go (hunk 5 above)
Comment (buptcozy) [reply pkg/webhook/elasticquota/validating/validating_handler.go:131]: fixed

File: pkg/webhook/api_debug_server.go (hunk 1 above)
Comment (buptcozy) [reply pkg/webhook/api_debug_server.go:31]: fixed

**Review by eahydra — COMMENTED** [review]
/lgtm

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Pull request merged**